kolibri/plugins/learn/assets/src/views/search-page/index.vue

```diff
@@ -1,149 +1,193 @@
-
-        <div v-if="noResults">{{ $tr('noResults') }}</div>
-
-        <div v-else>
-          <p>{{ $tr('results', {count: numResults}) }}</p>
-
-          <div v-if="topics.length">
-            <h2>{{ $tr('topics') }}</h2>
-            <card-list class="card-list">
-              <topic-list-item
-                v-for="topic in topics"
-                class="card"
-                :title="topic.title"
-                :link="genTopicLink(topic.id)"/>
-            </card-list>
-          </div>
-
-          <div v-if="contents.length">
-            <h2>{{ $tr('content') }}</h2>
-            <card-grid>
-              <content-grid-item
-                v-for="content in contents"
-                class="card"
-                :title="content.title"
-                :thumbnail="content.thumbnail"
-                :kind="content.kind"
-                :progress="content.progress"
-                :link="genContentLink(content.id)"/>
-            </card-grid>
-          </div>
-        </div>
-      </div>
-    </div>
+    <p v-if="!searchTerm">{{ $tr('noSearch') }}</p>
+
+    <template v-else>
+      <h1>{{ $tr('showingResultsFor', { searchTerm }) }}</h1>
+      <p>{{ $tr('withinChannel', { channelName }) }}</p>
+
+      <tabs>
+        <tab-button
+          type="icon-and-title"
+          :title="$tr('all', { num: all.length } )"
+          icon="layers"
+          @click="filter = 'all'"
+          :selected="filter === 'all'"
+        />
+        <tab-button
+          type="icon-and-title"
+          :title="$tr('topics', { num: topics.length } )"
+          icon="folder"
+          :selected="filter === contentNodeKinds.TOPIC"
+          @click="filter = contentNodeKinds.TOPIC"
+        />
+        <tab-button
+          type="icon-and-title"
+          :title="$tr('exercises', { num: exercises.length } )"
+          icon="star"
+          :selected="filter === contentNodeKinds.EXERCISE"
+          @click="filter = contentNodeKinds.EXERCISE"
+        />
+        <tab-button
+          type="icon-and-title"
+          :title="$tr('videos', { num: videos.length } )"
+          icon="ondemand_video"
+          :selected="filter === contentNodeKinds.VIDEO"
+          @click="filter = contentNodeKinds.VIDEO"
+        />
+        <tab-button
+          type="icon-and-title"
+          :title="$tr('documents', { num: documents.length } )"
+          icon="description"
+          :selected="filter === contentNodeKinds.DOCUMENT"
+          @click="filter = contentNodeKinds.DOCUMENT"
+        />
+        <tab-button
+          type="icon-and-title"
+          icon="widgets"
+          :title="$tr('html5', { num: html5.length } )"
+          :selected="filter === contentNodeKinds.HTML5"
+          @click="filter = contentNodeKinds.HTML5"
+        />
+      </tabs>
+
+      <p v-if="filteredResults.length === 0">
+        <template v-if="filter === 'all'">{{ $tr('noContent', { searchTerm }) }}</template>
```

Comment:
why not just make all of these options a computed prop that returns a translated string?